src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,39 @@
+import NextAuth, { Session } from "next-auth";
+import GoogleProvider from "next-auth/providers/google";
+import CredentialsProvider from "next-auth/providers/credentials"
+
+export const authOptions = {
+  providers: [
+    GoogleProvider({
+        clientId: process.env.GOOGLE_ID as string,
+        clientSecret: process.env.GOOGLE_SECRET as string
```

Comment:
to avoid using `as string` and also to have better error handling we should check for the important env vars before running the app, in this case google id and secret are important env vars so we want the app to throw an error if we forget to give them values

```suggestion
const { GOOGLE_ID, GOOGLE_SECRET } = process.env

if ( !GOOGLE_ID || !GOOGLE_SECRET ) {
  throw new Error('please provide google OAuth credentials')
}

export const authOptions = {
  providers: [
    GoogleProvider({
        clientId: GOOGLE_ID
        clientSecret: GOOGLE_SECRET
```